src/web/nextui/src/app/eval/ResultsCharts.tsx

```diff
@@ -415,13 +420,239 @@ function MetricChart({ table }: ChartProps) {
+        scales: {
+          x: {
+            title: {
+              display: true,
+              text: `Eval Index`,
+            },
+            type: 'linear',
+            position: 'bottom',
+            ticks: {
+              callback: function (value) {
+                if (Number.isInteger(value)) {
+                  return ordinalSuffixOf(Number(value));
+                }
+                return '';
+              },
+            },
+          },
+          y: {
+            title: {
+              display: true,
+              text: `Pass Rate`,
+            },
+            ticks: {
+              callback: function (value: string | number, index: number, values: any[]) {
+                let ret = String(Math.round(Number(value)));
+                if (index === values.length - 1) {
+                  ret += '%';
+                }
+                return ret;
+              },
+            },
+          },
+        },
+
+        plugins: {
+          legend: {
+            display: true,
+          },
+          tooltip: {
+            callbacks: {
+              title: function (context) {
+                return `Pass Rate: ${context[0].parsed.y.toFixed(2)}%`;
+              },
+              label: function (context) {
+                const point = context.raw as Point;
+                let label = point.metadata.label;
+                if (label && label.length > 30) {
+                  label = label.substring(0, 30) + '...';
+                }
+                return [
+                  `evalId: ${point.metadata.evalId}`,
+                  `Prompt: ${label}`,
+                  `Date: ${point.metadata.date}`,
+                  `Score: ${point.metadata.score}`,
+                ];
+              },
+            },
+          },
+        },
+
+        onClick: function (event, elements) {
+          if (elements.length > 0) {
+            const topMostElement = elements[0];
+            const pointData = (topMostElement.element as any).$context.raw as Point;
+            const evalId = pointData.metadata.evalId;
+            window.open(`/eval/?evalId=${evalId}`, '_blank');
+          }
+        },
+      },
+    });
+  }, [table, evalId, config, datasetId, dataset]);
+
+  return <canvas ref={lineCanvasRef} style={{ maxHeight: '300px', cursor: 'pointer' }} />;
+}
+
 function ResultsCharts({ columnVisibility }: ResultsChartsProps) {
   const theme = useTheme();
   Chart.defaults.color = theme.palette.mode === 'dark' ? '#aaa' : '#666';
   const [showCharts, setShowCharts] = useState(true);
+  const [showLineChart, setShowLineChart] = useState(false);
+
+  const { table, evalId, config } = useStore();
+
+  const datasetId = useMemo(() => {
+    if (config) {
+      return createHash('sha256').update(JSON.stringify(config.tests)).digest('hex');
+    }
+  }, [config]);
+
+  useMemo(async () => {
+    if (datasetId) {
+      const data = await fetchDataset(datasetId);
+      setShowLineChart(data.data[0].count > 1);
```

Comment:
On some existing ones, I'm getting `TypeError: Cannot read properties of undefined (reading 'count')` because the `/api/evals/:id` endpoint is returning `[]`

Reply:
adjusted, should fixed